src/api/index.js

```diff
@@ -2,69 +2,38 @@ import apiKey from './movie_api_key'
 
 // const dbUrl = 'http://localhost:3000/api/'
 const dbUrl = 'https://falsemotive.io/api/'
-const moviesUrl = 'https://api.themoviedb.org/3/movie/now_playing'
-
-// const options 
-
-export const postData = async (path = '', user) => {
-  const response = await fetch(`${dbUrl}${path}`, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(user)
-  })
-
-  if (response.ok) {
-    return response;
-  } else {
-    throw new Error(`${response.statusText}`);
-  }
+const moviesUrl = 'https://api.themoviedb.org/3/movie/'
+const options = (method, data) => ({
+  method,
+  headers: {
+    'Content-Type': 'application/json',
+  },
+  body: JSON.stringify(data)
+})
+
+export const postData = async (path = '', data) => {
+  const response = await fetch(`${dbUrl}${path}`, options('POST', data))
+  if (!response.ok) throw new Error(`${response.statusText}`)
+  const unfilteredData = await response.json()
+  return unfilteredData.data
 }
 
-export const postFavorite = async (path = '', user) => {
-  const response = await fetch(`${dbUrl}${path}`, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(user)
-  })
-
-  if (response.ok) {
-    return response;
-  } else {
-    throw new Error(`${response.statusText}`);
-  }
-}
-
-export const deleteFavorite = async (path = '', user) => {
-  const response = await fetch(`${dbUrl}${path}`, {
-    method: 'DELETE',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(user)
-  })
-
-  if (response.ok) {
-    return response;
-  } else {
-    throw new Error(`${response.statusText}`);
-  }
+export const deleteData = async (path = '', data) => {
+  const response = await fetch(`${dbUrl}${path}`, options('DELETE', data))
+  if (!response.ok) throw new Error(`${response.statusText}`)
+  const unfilteredData = await response.json()
+  return unfilteredData.results
 }
 
-export const fetchData = async (path) => {
+export const getData = async (path) => {
   const response = await fetch(`${dbUrl}${path}`)
-  if (response.ok) {
-    return await response.json()
-  } else {
-    throw new Error(`${response.message}: ${response.status}`);
-  }
+  if (!response.ok) throw new Error(`${response.statusText}`)
```

Comment:
This is a great refactor. It makes sense that we only need to check if the response is not ok, it's nice to see these function shortened up so much. They're also much more readable! 🔥